packages/kbn-i18n/src/angular/directive.ts

```diff
@@ -21,26 +21,37 @@ import { IDirective, IRootElementService, IScope } from 'angular';
 
 import { I18nServiceType } from './provider';
 
-export function i18nDirective(i18n: I18nServiceType): IDirective {
+interface I18nScope extends IScope {
+  values: any;
+  defaultMessage: string;
+  id: string;
+}
+
+export function i18nDirective(i18n: I18nServiceType): IDirective<I18nScope> {
   return {
     restrict: 'A',
     scope: {
       id: '@i18nId',
       defaultMessage: '@i18nDefaultMessage',
-      values: '=i18nValues',
+      values: '<i18nValues',
     },
-    link($scope: IScope, $element: IRootElementService) {
-      $scope.$watchGroup(
-        ['id', 'defaultMessage', 'values'],
-        ([id, defaultMessage = '', values = {}]) => {
-          $element.html(
-            i18n(id, {
-              values,
-              defaultMessage,
-            })
-          );
-        }
-      );
+    link($scope, $element) {
+      if ($scope.values) {
+        $scope.$watch('values', () => {
```

Comment:
__nit:__ I'm not sure how `values` are used in the Kibana currently, so wondering maybe we need `$watchCollection`? Or `$watch` is enough for our use case? @pavel06081991 @maryia-lapata @LeanidShutau